### ISSUE TITLE
SC assumes every address on outline is a entrance

### ISSUE THREAD

**Issue by ivanbranco** [streetcomplete/StreetComplete#4995]
Hi, as you already know in Italy we map addresses **only** as nodes, so this affects us particularly.

**How to Reproduce**
If you put an address (in the addresses layer) on a building outline, it presumes it's an entrance. This is not always the case, sometimes there's no entrance at all: 
![image](https://user-images.githubusercontent.com/94002623/236157583-c2168b1c-45cd-4099-a986-ab394e4d6bc6.png)
sometimes is just a shopping window:
![image](https://user-images.githubusercontent.com/94002623/236157687-7782694b-d838-4cea-8599-41dd2b86ee5e.png)

**Expected Behavior**
I guess a solution could be a toggle to enable/disable the "entrance" mode. Otherwise mapping would require an extra-step in another editor to check and fix all the addresses added (as I have now to do in iD). Also I would expect most users (the ones who are not "experts") to not even know about this, and not fixing them at all.

**Versions affected**
v53.0-beta1

**Comment by westnordost** [comment]
Wat. Uhm... I thought the whole point of the addressing system in Italy is that there is an address for every entrance...

How often does this occur? I presume they were entrances once, but not anymore. Maybe it could be an other-answer. I wouldn't want to litter the main UI with a checkbox like "yeah, there is a door here".

**Comment by westnordost** [comment]
ping @ivanbranco @Mannivu @Danysan1 @ricloy

**Comment by westnordost** [comment]
I tend to do nothing about this, because the user can just move the cursor away from the outline a bit if he just wants to add an address and not also an entrance. The UI should be clear about that an entrance is added when one places the address on the building outline itself.

**Comment by ivanbranco** [comment]
> ping @ivanbranco @Mannivu @Danysan1 @ricloy

Personally I don't have much more to add tbh, I do think this is a step back for the app based on my workflow. Before I used StreetComplete to collect addresses in the most precise way (so where there's physically the number plate if the entrance is missing), right now I switched to EveryDoor where I can snap the address to the building while deciding if it's a entrance or not.

I still do think this quest could bring to non-existent entrances, but that's just a guess of mine atm. I will try to keep an eye on italian changesets when this feature will come out of the beta and update this topic in case (even just for the records).

**Comment by dieterdreist** [comment]
I believe we should encourage the people to add entrance=no to these non-entrances which are permanently closed (also common with shop windows which are considered potential entrances regardless of there being a door or not)

**Comment by Danysan1** [comment]
> I thought the whole point of the addressing system in Italy is that there is an address for every entrance...
>
> How often does this occur? I presume they were entrances once, but not anymore. Maybe it could be an other-answer.

TLDR: Yes, in theory those should not have a house number. Their numbers may have been assigned before the current legislation was created or before the were closed.

I checked again the official document cited in [OSM Wiki](https://wiki.openstreetmap.org/wiki/IT:Addresses#Regole_specifiche_per_l'Italia) describing how cities should create civic numbers (["Informativa del 6 maggio 2014" by ISTAT](https://comune.messina.it/wp-content/plugins/allegati-rev2/includes/download.php?id=15294&ved=2ahUKEwi_ufLD4uP-AhX9gv0HHYDNBQoQFnoECAsQAQ&usg=AOvVaw2zFTndMRJysRMdTZfpkdVy), pp. 21-37).

At page 21 it states:

> Tutti gli accessi esterni di qualsiasi genere che dall’area di circolazione immettono direttamente o indirettamente all’interno dei fabbricati devono, secondo la normativa vigente, essere provvisti di appositi numeri civici

Translation:

> All external accesses of any kind that enter the buildings directly or indirectly from the circulation area must, according to current legislation, be provided with appropriate house numbers

Then at page 24 among the available exxamples I see this:

> ![image](https://user-images.githubusercontent.com/8406735/236695561-00e0f44f-ef4d-4a8a-9c75-d9ab9248b2f1.png)
>
> Casi in cui è prevista la numerazione civica
> a. Accesso diretto al box
> c. Vetrina a livello terra
> d. Accesso diretto al negozio
> e. Accesso indiretto a unità immobiliari interne all’edificio
> f. Vetrina a livello terra
> g. Vetrina e ingresso. L’insieme è considerato un unico accesso temporaneamente con due funzioni
> h. Vetrina e ingresso. L’insieme è considerato un unico accesso temporaneamente con due funzioni
> i. Accesso diretto ad abitazione
>
> Casi in cui non è prevista la numerazione civica
> b. Vetrina sollevata dal livello terra da un muretto superiore a 25 cm

Translation of the last part:

> Cases in which house numbering is not provided
> b. Showcase raised from the ground level by a wall higher than 25 cm

This is reinforced at page 29:

> Gli accessi temporaneamente chiusi da vetrine o da altro materiale (non in muratura) vanno numerati se 
posti a livello terra o su un gradino di alzata massima di 25 cm

Translation:

> Accesses temporarily closed by display cases or other material (not masonry) must be numbered if located at ground level or on a step with a maximum height of 25 cm

Accesses closed by masonry, like the first example in this issue, are not cited directly, so applying the general rule I would say that in theory it shouldn't have a number.
I would say that the second example in the issue follows the 25cm rule and in theory shouldn't have anumber.
Maybe they have been assigned their own number before the current law was created or before the were closed.

-----

> Are addresses without entrances actually mapped as vertices on the outline of buildings?

Yes, as can be seen [by searching elements with addr:housenumber=* and entrance=no](http://overpass-turbo.eu/s/1uDN).
Does this make sense? TLDR: IMO yes

* Currently [the scheme for Italy in the OSM Wiki](https://wiki.openstreetmap.org/wiki/IT:Addresses#Regole_specifiche_per_l'Italia), which tries to follow the housenumber as defined by the law, says to map the address on the entrance and does not address specifically this situation, so following this guideline it makes sense to map on the outline.
* If we want to evaluate the source of this scheme, the law, it doesn't help us because it defines the meaning of numbers for entrances, when there is a number without entrance it doesn't help us anymore
* As a fallback I would say to "map what's on the ground" by mapping the house number where it is actually placed: in these cases the housenumber is physically placed on the real-world outline of the building so it makes sense

-----

> I wouldn't want to litter the main UI with a checkbox like "yeah, there is a door here".

This is a good intention, but not inserting wrong data in OSM should have the priority. I would say the options are:
* asking whether there is really an entrance (given what we said above this would be the perfect solution)
* make absolutely sure the user knows that when he is adding the address he is also adding the entrance (this would be less heavy on the UX but it would prevent from adding non-entrance addresses on the outline)
* never add entrance=* (this would be like splitting the baby... not ideal)

> I tend to do nothing about this, because the user can just move the cursor away from the outline a bit if he just wants to add an address and not also an entrance. The UI should be clear about that an entrance is added when one places the address on the building outline itself.

So the second option cited above. This is not perfect but I guess it could be considered an acceptable compromise, as long as the app is absolutely sure the user knows that he is adding an `entrance=*`. Currently there is no warning of this, not even only on ther first time, just the symbol of the door in the marker (right?), which is not clear enough.

**Comment by westnordost** [comment]
Thank you @Danysan1. So to summarize, indeed, it is actually quite common in Italy: Not only entrances have housenumbers, but (theoretically?) every single shop window and also every garage gate.

We want to avoid that `entrance=yes` is added in error. Also, if possible, we want to be able to still add the address to the building outline as a vertex without adding `entrance=yes` where it is indeed not an entrance. At the same time, we do want to add `entrance=yes` to entrances with housenumbers.

I thought of the following solutions, see my last paragraph by which I decided to go with:

1. never add `entrance=yes` (in Italy). The possible error of adding `entrance=yes` is eliminated, but at the cost of not adding entrances at all. This solution is only really eligible if one could say that the current UI is somewhat mistakable; which it isn't, IMO, otherwise OP wouldn't have realized what the behavior in the beta is. (see here for a video: https://github.com/streetcomplete/StreetComplete/releases/tag/v53.0-beta1) I think anyone understands what a door is.

2. a UI element with which to add `entrance=yes`/`no`, maybe only for Italy. There are various reasons against this: First, a checkbox won't do, it'd have to be either a radio button group `Is an entrance: ( ) yes ( ) no` or a switch `<=O> is entrance` - the latter is a little weird, because a switch is usually used to turn something on and off, not to answer a yes/no question -> possible misunderstanding. Second, it clutters the UI. The UI for the address is already quite large. Third, it's not even as easy as that. A garage door (`barrier=gate`) has an address too. So, it'd have to be a 3-option-select... or more (`barrier=lift_gate`?, ...)

3. add "other answer" option "Not an entrance" to turn the door icon into the address icon for the currently open form. This has the following problems: Apparently in Italy **a lot** of addresses are not entrances (e.g garage doors, some shop windows), so Italian users would need to tap that button a lot. Additionally, what to tag in that case, `entrance=no` won't work because see point 2, so it needs to be left empty. This is not ideal, because now the information is still missing if it is an entrance or not, it's undefined.

So, from these, I prefer option 3 because it is least problematic, however I think still the best solution is to do... (almost) nothing:

If you want to place an address that is not an entrance, just move the cursor a bit away from the building outline so that it does not snap to the outline anymore. As for option 3, it is still somewhat undefined whether it is an entrance or not, or a garage door or whatever, but IMO it is less inviting to just smack a `entrance=yes` on it for other mappers - because it is not even on the outline. However, what I could and should do is to reduce the snap distance a bit, so that one doesn't need to move the mouse cursor so far towards the building center to not have it snap.

**Comment by dieterdreist** [comment]
sent from a phone

> On 8 May 2023, at 00:12, Tobias Zwick ***@***.***> wrote:
> 
> A garage door (barrier=gate) has an address too.


a barrier=gate is not a garage door usually/often, it is a gate in a fence or wall, either a building passage or the access to a plot (could be a gate for pedestrians, or a larger one for vehicles, also for gates leading to gardens)

**Comment by westnordost** [comment]
Or `amenity=parking_entrance`. But..., just proves my point (for point 2).

**Comment by dieterdreist** [comment]
or barrier=entrance 
these are more or less all the possibilities, if SC could ask about this kind of information for housenumbers it would be great

**Comment by mnalis** [comment]
> a barrier=gate is not a garage door usually/often,

> Or amenity=parking_entrance

Or [entrance=garage](https://wiki.openstreetmap.org/wiki/Tag:entrance=garage) ?

But anyway, to me it also seems that current solution (but with somewhat reduced snap-width) is good enough; the change of `address icon` to `door icon` looks quite discoverable to me at least. 

Of course, that only works if the Italian community is OK with address being inside a building a little in cases when there is not entrance?

**Comment by Danysan1** [comment]
> if one could say that the current UI is somewhat mistakable; which it isn't, IMO, otherwise OP wouldn't have realized what the behavior in the beta is and written this ticket. (see here for a video how it looks https://github.com/streetcomplete/StreetComplete/releases/tag/v53.0-beta1) I find it hard to believe that the door icon is ambiguous. Everyone understands what a door is.

Yes, I referred to that icon. Everyone understands what a door is but IMO it's not obvious that it means also explicitly making the address as an entrance (or even knowing this is possible in OSM).
OP is an expert OSM mapper who knows well how to look at his past changesets and which tags to expect. Unless the target audience of the app includes only users with this level of knowledge, it is likely that more inexperienced users will not recognize the error when they make it.
This would be much more clear if a confirmation pop-up would appear only the first time the user inserts an entrance warning him/her that he/she is explicitly making the address as an entrance.

**Comment by matkoniecz** [comment]
> IMO it's not obvious that it means also explicitly making the address as an entrance

How else door icon could be understood?

**Comment by matkoniecz** [comment]
I am asking about something else: is there an actual viable way to misunderstood this door icon?

**Comment by Danysan1** [comment]
As said above IMO it's not obvious that it means explicitly making the address as an entrance, but I guess it doesn't matter anymore now

**Comment by waldyrious** [comment]
> EveryDoor where I can snap the address to the building

@ivanbranco wait, EveryDoor allows snapping entrances to building outlines? How does one do that?

**Comment by Mannivu** [comment]
> How does one do that?

@waldyrious EveryDoor does it automatically if you add a node close to a line (sometimes is the right line, sometimes is not).

**Comment by mnalis** [comment]
> As said above IMO it's not obvious that it means explicitly making the address as an entrance, 

@Danysan1 While I think it was clear that the `door` icon did not symbolize to you that the **entrance** is going to be placed there, the request for clarification was I think this:

- _"what **does** the `door` icon symbolize **to you** in that context"_? (as it does **not** seem to symbolize entrance)
- Or did you not even notice the icon changes?
- Or you noticed the icon change, but didn't pay it any attention as you assumed it is just a visual gimmick without any meaning?
- Or you noticed the icon change, but were utterly confused why it did so and what it might mean?
- Or something else (what?)

> but I guess it doesn't matter anymore now

It would still provide **valuable feedback** to developers, so they can get a feeling what kind of UI works good, and what kind not so good; which would be helpful for further quests/overlays too. So it would be helpful if you can provide us with information why that UI was unintuitive to you (and, if you have idea perhaps, what kind of User Interface would have been more understandable to you?)

**Comment by dieterdreist** [comment]
I think setting entrance=yes/no/main should be encouraged by the app

**Comment by Danysan1** [comment]
> "what does the door icon symbolize to you in that context"?

In short, it's the second point you cited: 
I noticed the icon change and having a decent knowledge of OSM's tagging schemes I imagined it could also represent the addition of an entrance but
- given the context it's obvious that an address will be added but the same does not apply for the entrances, a newby user could simply assume it is just a visual gimmick without any meaning
- a newby user may not even know OSM can also map entrances
- even if you imagine the logo could represent an `entrance=*` the only way to know for sure whether it will really add an entrance or not is by actually inserting the address and checking on OSM the new element

Sure, it's not a major bug, but it could be a user experience improvement


> if you have idea perhaps, what kind of User Interface would have been more understandable to you?

Thinking to a solution which is clear but not too invasive, the first idea that comes to my mind is the confirmation pop-up I talked above that appears only the first time the user inserts an entrance warning him/her that he/she is explicitly making the address as an entrance.

**Comment by zlita** [comment]
I would argue that the implementation of the option 3 is insufficient.

I recently discovered by pure luck that adding an address close to a building also add an entrance. I've previously mapped hundreds of address in France (where, most address aren't located at entrances) without discovering this fact.

From my point of view the current implementation raise two problems :
- It is not explicit at all that the created address is linked to an entrance. The only way to discover it is to see that there is two different icon's color (which will not be the case if every created address is close to a building) and, once you've seen it, you have to look at the tags to understand what is happening (and be able to understand that there is an entrance where there shouldn't be one). This problem may (and in fact in my case it did happen) lead to creating entrances where there is not one.
- As previously said in the discussion, once you've discovered this, you have to manually tap the "this is not an entrance" button every time despite the fact that it is in a menu. It is not dangerous like the discoverability issue, but it makes the process of mapping address way more tedious. I think that, in countries where address and entrances aren't always located on the same place, either not setting the entrance tag or adding a visible toggle on the ui would be better. The advantage of the toggle button would be to also make the association explicit, which would solve the discoverability problem.


Another option to solve the discoverability problem might be to change the icon to an "entry icon" when the entrance is linked to an entry. However, this option doesn't help to discover that you can disable this link (and I'm not sure that a beginner mapper will understand that it lead to a difference in the mapping).

**Comment by matkoniecz** [comment]
> I would argue that the implementation of the option 3 is insufficient.

iff problem still exists, please create a new issue

comment in issue from 2023 is likely to be missed/forgotten

> It is not explicit at all that the created address is linked to an entrance

IIRC you get dedicated door icon in address adder
